orderer/common/cluster/commauth_test.go

```diff
@@ -0,0 +1,431 @@
+	go cn.srv.Start()
+}
+
+func newClusterNodeWithMetrics(t *testing.T, metrics cluster.MetricsProvider, tlsConnGauge metrics.Gauge) *clusterServiceNode {
+	serverKeyPair, err := ca.NewServerCertKeyPair("127.0.0.1")
+	require.NoError(t, err)
+
+	clientKeyPair, _ := ca.NewClientCertKeyPair()
+
+	handler := &mocks.Handler{}
+	clientConfig := comm_utils.ClientConfig{
+		AsyncConnect: true,
+		DialTimeout:  time.Hour,
+		SecOpts: comm_utils.SecureOptions{
+			RequireClientCert: true,
+			Key:               clientKeyPair.Key,
+			Certificate:       clientKeyPair.Cert,
+			ServerRootCAs:     [][]byte{ca.CertBytes()},
+			UseTLS:            true,
+			ClientRootCAs:     [][]byte{ca.CertBytes()},
+		},
+	}
+
+	dialer := &cluster.PredicateDialer{
+		Config: clientConfig,
+	}
+
+	srvConfig := comm_utils.ServerConfig{
+		SecOpts: comm_utils.SecureOptions{
+			Key:         serverKeyPair.Key,
+			Certificate: serverKeyPair.Cert,
+			UseTLS:      true,
+		},
+	}
+	gRPCServer, err := comm_utils.NewGRPCServer("127.0.0.1:", srvConfig)
+	require.NoError(t, err)
+
+	tstSrv := &clusterServiceNode{
+		dialer:       dialer,
+		clientConfig: clientConfig,
+		serverConfig: srvConfig,
+		bindAddress:  gRPCServer.Address(),
+		handler:      handler,
+		nodeInfo: cluster.RemoteNode{
+			Endpoint:      gRPCServer.Address(),
+			ID:            nextUnusedID(),
+			ServerTLSCert: serverKeyPair.TLSCert.Raw,
+			ClientTLSCert: clientKeyPair.TLSCert.Raw,
+		},
+		srv: gRPCServer,
+	}
+
+	compareCert := cluster.CachePublicKeyComparisons(func(a, b []byte) bool {
+		return crypto.CertificatesWithSamePublicKey(a, b) == nil
+	})
+
+	if tstSrv.dispatcher == nil {
+		tstSrv.dispatcher = tstSrv
+	}
+
+	tstSrv.c = &cluster.Comm{
+		CertExpWarningThreshold: time.Hour,
+		SendBufferSize:          1,
+		Logger:                  flogging.MustGetLogger("test"),
+		Chan2Members:            make(cluster.MembersByChannel),
+		H:                       handler,
+		ChanExt:                 channelExtractor,
+		Connections:             cluster.NewConnectionStore(dialer, tlsConnGauge),
+		Metrics:                 cluster.NewMetrics(metrics),
+		CompareCertificate:      compareCert,
+		ServiceClientVersion:    cluster.ClusterServiceClientWithAuth,
+	}
+
+	orderer.RegisterClusterNodeServiceServer(gRPCServer.Server(), tstSrv.dispatcher)
+	go gRPCServer.Start()
+	return tstSrv
+}
+
+func newTestNodeWithAuthentication(t *testing.T) *clusterServiceNode {
+	return newClusterNodeWithMetrics(t, &disabled.Provider{}, &disabled.Gauge{})
+}
+
+func TestCommServiceBasicAuth(t *testing.T) {
+	// Scenario: Basic test that spawns 2 nodes and sends each other
+	// messages that are expected to be echoed back
+
+	node1 := newTestNodeWithAuthentication(t)
+	node2 := newTestNodeWithAuthentication(t)
+
+	defer node1.stop()
+	defer node2.stop()
+
+	config := []cluster.RemoteNode{node1.nodeInfo, node2.nodeInfo}
+	node1.c.Configure(testChannel, config)
```

Comment:
when you call `Configure`, the code still tries to use the TLS certificates for egress. This is now how it's supposed to work. We no longer use TLS pinning now.